### ISSUE TITLE
@gitbeaker/cli does not work on Node 22

### ISSUE THREAD

**Issue by kashz** [jdalrymple/gitbeaker#3591]
**Description**

- Node.js version: 22.2.0
- Gitbeaker version: 40.0.3
- Gitbeaker release (cli, rest, core, requester-utils): core, cli

```
$ gitbeaker --version
file:///Users/xxx/.asdf/installs/nodejs/22.2.0/lib/node_modules/@gitbeaker/cli/dist/index.mjs:5
import API_MAP from '@gitbeaker/core/map.json' assert { type: 'json' };
                                               ^^^^^^

SyntaxError: Unexpected identifier 'assert'
    at compileSourceTextModule (node:internal/modules/esm/utils:337:16)
    at ModuleLoader.moduleStrategy (node:internal/modules/esm/translators:163:18)
    at callTranslator (node:internal/modules/esm/loader:430:14)
    at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:436:30)
    at async ModuleJob._link (node:internal/modules/esm/module_job:106:19)

Node.js v22.2.0
```

I've encountered an issue where @gitbeaker/cli does not work with Node 22.
This is because the `assert` keyword has been removed in Node 22.

https://github.com/nodejs/node/pull/52104

Replacing it with the `with` keyword should resolve the issue, but the `with` keyword does not work on Node versions less than 18.20.0, which may affect the supported Node versions.

- https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-3.html#import-attributes
- https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V18.md#added-support-for-import-attributes

**Checklist**

- [x] I have checked that this is not a duplicate issue.
- [x] I have read the documentation.

**Comment by jdalrymple** [comment]
Shouldnt be a problem since the libs lowest version is 18!

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`40.1.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/40.1.0) :rocket:

**Comment by kashz** [comment]
@jdalrymple
Thank you for your work on this issue. However, it seems that the issue has not been fixed yet in version 40.1.2.
The `with` keyword is being converted to `assert` after the build process.

https://www.npmjs.com/package/@gitbeaker/cli/v/40.1.2?activeTab=code

This happens because when `tsup` is used with the `--treeshake` option, it uses `rollup` for bundling instead of `esbuild`.

https://github.com/jdalrymple/gitbeaker/blob/40.1.2/packages/cli/package.json#L34
https://tsup.egoist.dev/#tree-shaking

By default, `rollup` converts import attributes to `assert`.

https://rollupjs.org/configuration-options/#output-importattributeskey

Since `esbuild` performs tree shaking by default, I think using `esbuild` might be a better solution.
What are your thoughts on this?

**Comment by jdalrymple** [comment]
I really thought tsup used esbuild under the hood. Im open to using instead!
